apps/web/src/app/HSL-API-endpoint/fetch-stops/route.ts

```diff
@@ -0,0 +1,210 @@
+import { ApolloClient, gql, InMemoryCache } from "@apollo/client";
+import type { ApolloQueryResult, DocumentNode } from "@apollo/client";
+import type {
+  Arrival,
+  ArrivalAttribute,
+  RenderableStop,
+  Stop,
+  StopOutData,
+  StopTime,
+  StopType,
+  Data,
+  StopFromApi,
+} from "../../../assets/HSL-API-types/hsl-helper-types";
+
+const STOPS = [
+  ["HSL:2222406", "HSL:2222405"],
+  ["HSL:2222603", "HSL:2222604"],
+  ["HSL:2222234", "HSL:2222212"],
+]; // [["HSL:2222234", "HSL:E22259"], ["HSL:2000102"], ["HSL:E0773", "HSL:E0772"]];
+
+export const dynamic = "force-dynamic";
+
+export async function GET() {
+  const dataFromHsl: RenderableStop[] = [];
+  for (const stop of STOPS) {
+    const stp: RenderableStop | null = await getStop(stop);
+    if (stp) {
+      dataFromHsl.push(stp);
+    }
+  }
+
+  const retData = {
+    type: "Data",
+    data: dataFromHsl,
+  };
+
+  return Response.json({ retData });
+}
+
+const GetStopSchedule = (StopId: string): DocumentNode =>
+  gql(`
+  {
+    stop(id: "${StopId}") {
+      name
+        stoptimesWithoutPatterns {
+        realtimeArrival
+        realtimeArrival
+        serviceDay
+        headsign
+        trip{
+          routeShortName
+        }
+      }
+    }
+  }
+`);
+
+function pad(number: number, size: number) {
+  let s = String(number);
+  while (s.length < (size || 2)) {
+    s = "0".concat(s);
+  }
+  return s;
+}
+
+function removeSubstring(fullString: string): string {
+  const subStrings: string[] = [
+    " via Leppävaara",
+    " via Rautatientori",
+    " via Tapiola (M)",
+    " via Huopalahti as.",
+    " via Tapiola",
+    " via Pasila as.",
+  ];
+  let str = "";
+  for (const subString of subStrings) {
+    str = fullString.replace(subString, "");
+  }
+  return str;
+}
+
+function isTram(arrival: Arrival): boolean {
+  return arrival.route.includes("15");
+}
+function isMetro(arrival: Arrival): boolean {
+  return arrival.route.includes("M");
```

Comment:
This seems a bit flaky? What if the route is Matinkylä or something else with a capital M

Reply:
The arrival.route only includes the number of the bus i.e. 108N, 111, 52, 510 etc. so the M is only reserved for thee metro line.
arrival.headsign includes e.g. matinkylä and such. In the case the metro is replaced by a bus